ruby/lib/asciidoctor/extensions/asciidoctor_kroki/extension.rb

```diff
@@ -252,10 +252,8 @@ def output_dir_path(doc)
         images_dir = doc.attr('imagesdir', '')
         if (images_output_dir = doc.attr('imagesoutdir'))
           images_output_dir
-        elsif (out_dir = doc.attr('outdir'))
+        elsif (out_dir = doc.attr('outdir') || (doc.nested? ? doc.parent_document : doc).options[:to_dir])
```

Comment:
I've patched core so extensions don't have to worry about the nested document, if you prefer to wait for a release. See https://github.com/asciidoctor/asciidoctor/commit/7edc9da023522be67b17e2a085d72e056703a438

Reply:
I think it's safer to keep it a bit longer but we should certainly add a comment to state that this is unnecessary with Asciidoctor >= 2.0.18